src/main/kotlin/org/malachite/estella/aplication/api/ApplicationController.kt

```diff
@@ -59,6 +65,32 @@ class ApplicationController(
     fun deleteApplication(@PathVariable applicationId: Int) =
         applicationService.deleteApplication(applicationId)
 
+    @CrossOrigin
+    @PutMapping("/{applicationId}/next")
+    fun updateApplicationStage(@RequestHeader(EStellaHeaders.jwtToken) jwt: String?, @PathVariable applicationId: Int): ResponseEntity<Message> {
+        applicationService.getApplicationById(applicationId).let {
+            recruitmentProcessService.getProcessFromStage(it.stage).let {
+                if (!checkUserRights(it.offer, jwt)) return OwnResponses.UNAUTH
+                return applicationService.setNextStageOfApplication(applicationId).let { OwnResponses.SUCCESS }
+            }
+        }
+    }
+
+    @CrossOrigin
+    @PutMapping("/{applicationId}/reject")
+    fun rejectApplication(@RequestHeader(EStellaHeaders.jwtToken) jwt: String?, @PathVariable applicationId: Int): ResponseEntity<Message> {
+        applicationService.getApplicationById(applicationId).let {
+            recruitmentProcessService.getProcessFromStage(it.stage).let {
+                if (!checkUserRights(it.offer, jwt)) return OwnResponses.UNAUTH
+                return applicationService.rejectApplication(applicationId).let { OwnResponses.SUCCESS }
+
+            }
+        }
+    }
+
+    private fun checkUserRights(offer: Offer, jwt: String?) =
+        securityService.checkOfferRights(offer, jwt)
+
     @ExceptionHandler(NoSuchElementException::class)
```

Comment:
Wiem, że to nie Twoje, ale wyniósłbym exception handlera do osobnego pliku - będzie nam się wtedy lepiej szukało, jeśli konwencja będzie taka sama wszędzie